### PR TITLE
Cleanup compiler warnings in strict environment.

### PR DESCRIPTION
**Pull request by h2zero**
This clears up many compilation issues when using -Wall -Werror=all -Wextra options.

### CODE REVIEW

File: main/ZgatewayBT.ino [hunk 1]
```diff
@@ -1029,7 +1035,7 @@ void launchBTDiscovery() {
       String macWOdots = String(p->macAdr);
       macWOdots.replace(":", "");
       Log.trace(F("Launching discovery of %s" CR), p->macAdr);
-      if (p->sensorModel == HHCCJCY01HHCC) MiFloraDiscovery((char*)macWOdots.c_str(), "HHCCJCY01HHCC");
```
Comment (1technophile) [inline main/ZgatewayBT.ino:1032]: I'm not sure why the miflora doesn't need the (char *) here, should it be removed for all discovery calls?

Comment (h2zero) [reply main/ZgatewayBT.ino:1032]: Yes, that does seem odd doesn't it. The reason was I simply did a quick compile check with just one and forgot to go back and change the rest. I'll remove the others and push the change.

Comment (h2zero) [reply main/ZgatewayBT.ino:1032]: Done

**Comment by 1technophile** [comment]
Thanks!

**Pull request merged**